src/AbstractFactory/ReflectionBasedAbstractFactory.php

```
@@ -0,0 +1,188 @@
+     */
+    public function __invoke(ContainerInterface $container, $requestedName, array $options = null)
+    {
+        $reflectionClass = new ReflectionClass($requestedName);
+
+        if (null === ($constructor = $reflectionClass->getConstructor())) {
+            return new $requestedName();
+        }
+
+        $reflectionParameters = $constructor->getParameters();
+
+        if (empty($reflectionParameters)) {
+            return new $requestedName();
+        }
+
+        $parameters = array_map(
+            $this->resolveParameter($container, $requestedName),
+            $reflectionParameters
+        );
+
+        return new $requestedName(...$parameters);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function canCreate(ContainerInterface $container, $requestedName)
+    {
+        return class_exists($requestedName);
+    }
+
+    /**
+     * Resolve a parameter to a value.
+     *
+     * Returns a callback for resolving a parameter to a value.
+     *
+     * @param ContainerInterface $container
+     * @param string $requestedName
+     * @return callable
+     */
+    private function resolveParameter(ContainerInterface $container, $requestedName)
+    {
+        $hasConfig = $container->has('config');
+
+        /**
+         * @param ReflectionClass $parameter
+         * @return mixed
+         * @throws ServiceNotFoundException If type-hinted parameter cannot be
+         *   resolved to a service in the container.
+         */
+        return function (ReflectionParameter $parameter) use ($container, $requestedName, $hasConfig) {
+            if ($parameter->isArray()
+                && $parameter->getName() === 'config'
+                && $hasConfig
+            ) {
+                return $container->get('config');
+            }
+
+            if ($parameter->isArray()) {
+                return [];
+            }
+
+            if (! $parameter->getClass()) {
+                return;
```

Here I'd expect it to throw an exception as well instead of failing silently.